packages/core/saved-objects/core-saved-objects-utils-server/src/saved_objects_test_utils.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { arrayMapsAreEqual, setMapsAreEqual, setsAreEqual } from './saved_objects_test_utils';
+
+describe('savedObjects/testUtils', () => {
+  describe('#setsAreEqual', () => {
+    const setA = new Set(['1', '2', '3']);
+    const setB = new Set(['1', '2']);
+    const setC = new Set(['1', '3', '4']);
+    const setD = new Set(['2', '3', '1']);
+
+    describe('inequal', () => {
+      it('should return false is the sets are not the same size', () => {
+        expect(setsAreEqual(setA, setB)).toBeFalsy();
+        expect(setsAreEqual(setB, setA)).toBeFalsy();
+        expect(setsAreEqual(setA, new Set())).toBeFalsy();
+        expect(setsAreEqual(new Set(), setA)).toBeFalsy();
+      });
+
+      it('should return false is the sets do not have the same values', () => {
+        expect(setsAreEqual(setA, setC)).toBeFalsy();
+        expect(setsAreEqual(setC, setA)).toBeFalsy();
+      });
+    });
+
+    describe('equal', () => {
+      it('should return true if the sets are exactly the same', () => {
+        expect(setsAreEqual(setA, setD)).toBeTruthy();
+        expect(setsAreEqual(setD, setA)).toBeTruthy();
+        expect(setsAreEqual(new Set(), new Set())).toBeTruthy();
+      });
+    });
+  });
+
+  describe('#arrayMapsAreEqual', () => {
```

Comment:
__nit:__ it looks like we support both `undefined` and `string[]` for values, can have a few more test cases with these values too (`undefined` and empty arrays too)?